src/js/core/widget/core/OnOffSwitch.js

```diff
@@ -386,6 +399,136 @@
+			}
+
+			OnOffSwitch.prototype._onDragEnd = function (event) {
+				var self = this,
+					ui = self._ui,
+					checkbox = ui.input,
+					handler = ui.handler,
+					moveWidth = self._moveWidth,
+					initialPos = checkbox.checked ? moveWidth : 0,
+					currentPos,
+					checkedByPosition;
+
+				currentPos = initialPos + event.detail.deltaX;
+				currentPos = Math.min(Math.max(currentPos, 0), moveWidth);
+
+				handler.classList.remove(classes.onDrag);
+
+				checkedByPosition = currentPos > (moveWidth / 2);
+
+				// add animation
+				if (checkedByPosition) {
+					handler.classList.add(classes.moveToOn);
+				} else {
+					handler.classList.add(classes.moveToOff);
+				}
+				// remove temporary styles from drag
+				ui.input.style.backgroundColor = null;
+				ui.input.style.borderColor = null;
+				ui.input.style.borderWidth = null;
+
+				if (checkedByPosition !== checkbox.checked) {
+					// toggle on-off switch
+					self._setValue(checkedByPosition ? 1 : 0);
+
+					// trigger change event
+					onChangeValue(self);
+				}
+			}
+
+			function calculateColor(data, progress) {
+				var from = data.from,
+					to = data.to;
+
+				return [
+					progress * Math.abs(to.r - from.r),
+					progress * Math.abs(to.g - from.g),
+					progress * Math.abs(to.b - from.b),
+					progress * Math.abs(to.a - from.a)
+				];
+			}
+
+			OnOffSwitch.prototype._onDrag = function (event) {
+				var self = this,
+					ui = self._ui,
+					moveWidth = self._moveWidth,
+					initialPos = ui.input.checked ? moveWidth : 0,
+					currentPos,
+					progress;
+
+				currentPos = initialPos + event.detail.deltaX;
+				currentPos = Math.min(Math.max(currentPos, 0), moveWidth);
+
+				ui.handler.style.transform = "translateX(" + currentPos + "px)";
+
+				progress = moveWidth ? currentPos / moveWidth : 1;
+				ui.input.style.backgroundColor = "rgba(" +
+					calculateColor(self._transform.bgColor, progress).join(",") +
+					")";
+				ui.input.style.borderColor = "rgba(" +
+					calculateColor(self._transform.borderColor, 1 - progress).join(",") +
+					")";
+				ui.input.style.borderWidth = (1 - progress) + "px";
+			};
+
+			OnOffSwitch.prototype._onAnimationEnd = function (event) {
+				var targetClassList = event.target.classList;
+
+				targetClassList.remove(classes.moveToOff);
+				targetClassList.remove(classes.moveToOn);
+				this._ui.handler.style.transform = null;
+			};
+
+			OnOffSwitch.prototype.handleEvent = function (event) {
+				var self = this;
+
+				switch (event.type) {
+					case "change":
+						onChangeValue(self);
+						break;
+					case "focus":
+						self._focus(event);
+						break;
+					case "blur":
+						self._blur(event);
+						break;
+					case "keyup":
+						self._keyUp(event);
+						break;
+					case "drag":
+						self._onDrag(event);
+						break;
+					case "dragstart":
+						self._onDragStart(event);
+						break;
+					case "dragend":
+						self._onDragEnd(event);
+						break;
+					case "animationend":
```

Comment:
Do you use both "end" and "End"?

Reply:
Yes, it is intended. It happened that some versions of the web engine on  Tizen generated the event with this name.